bofire/data_models/strategies/predictives/botorch.py

```diff
@@ -32,6 +34,184 @@
+            used to avoid memory issues.
+        experiment_batch_sizes: List of sizes of the experiment batches for
+            each trust region. Used to determine whether or not a batch led to
+            a success or failure and to prevent data sharing between trust regions.
+        use_independent_tr: If true, each trust region will only fit to data
+            from its own history. If false (default), all trust regions will
+            fit to available data that overlaps the trust region.
+    """
+
+    type: str
+    length_init: float = 0.8
+    length_min: float = 0.01
+    length_max: float = 1.6
+    lengthscale_adjustment_factor: float = 2.0
+    fit_region_multiplier: float = 2.0
+    min_tr_size: PositiveInt = 10
+    max_tr_size: PositiveInt = 2048
+    success_epsilon: float = 1e-3
+    success_streak: PositiveInt = 3
+    failure_streak: PositiveInt = 3
+    success_counter: PositiveInt = 0
+    failure_counter: PositiveInt = 0
+    experiment_batch_sizes: list[list[PositiveInt]] = Field(
+        default_factory=lambda: [[]]
+    )
+    use_independent_tr: bool = False
+
+    length: float = Field(default=length_init, alias="length_")
+    X_center_idx: int = -1
+
+    @model_validator(mode="after")
+    def validate_fit_region_multiplier(self):
+        if self.fit_region_multiplier < 1:
+            raise ValueError("fit_region_multiplier must be >= 1.")
+        return self
+
+    @abstractmethod
+    def update_trust_region(self, experiments: pd.DataFrame, domain: Domain) -> None:
+        """Method to update the trust region based on the success of the optimization step.
+
+        Args:
+            experiments (pd.DataFrame): DataFrame containing experiments that were
+                performed and their results.
+            domain (Domain): domain containing the inputs and constraints.
+        """
+
+    def has_sufficient_experiments(
+        self, experiments: pd.DataFrame, domain: Domain
+    ) -> bool:
+        """Do we have enough samples in the trust region to fit a model?
+
+        Args:
+            experiments (pd.DataFrame): DataFrame containing experiments that
+                were performed and their results.
+            domain (Domain): The domain defining the problem to be optimized
+                with the strategy.
+        """
+        return (
+            self.min_tr_size
+            < self.get_trust_region_experiments(experiments, domain).shape[0]
+        )
+
+    def init_trust_region(self, experiments: pd.DataFrame, domain: Domain) -> None:
+        """Method to initialize the trust region.
+
+        Args:
+            experiments (pd.DataFrame): DataFrame containing experiments that
+                were performed and their results.
+            domain (Domain): The domain defining the problem to be optimized
+                with the strategy.
+        """
+
+        self.length = self.length_init
+
+        Y_cols = domain.outputs.get_keys()
+        if len(Y_cols) != 1:
```

Comment:
Ah ok, makes sense. Perhaps one could then also inherit it just from SoboStrategy?

Reply:
Or build in there?

Reply:
This should probably be an abstract method here and then move this init to the `TuRBOConfig`

Reply:
Yeah, I think this would make sense to have it as abstract method, especially as there and move it to `TuRBOConfig`, expecially as the there are TR methods that support multiobjective optimization.